server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -1038,26 +1040,91 @@ func (s *BuildBuddyServer) getAnyAPIKeyForInvocation(ctx context.Context, invoca
 	return apiKeys[0], nil
 }
 
-// Handle requests for build logs and artifacts by looking them up in from our
-// cache servers using the bytestream API.
+// ServeHTTP handles requests for build logs and artifacts either by looking
+// them up from our cache servers using the bytestream API or pulling them
+// from blobstore.
 func (s *BuildBuddyServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	params := r.URL.Query()
+
+	// TODO(siggisim): Figure out why this JWT is overriding authority auth and remove.
+	ctx := context.WithValue(r.Context(), "x-buildbuddy-jwt", nil)
+
+	if params.Get("artifact") != "" {
+		s.serveArtifact(ctx, w, params)
+		return
+	}
+	if params.Get("bytestream_url") != "" {
+		// bytestream request
+		s.serveBytestream(ctx, w, params)
+		return
+	}
+}
+
+// serveArtifact handles requests that specify particular build artifacts
+func (s *BuildBuddyServer) serveArtifact(ctx context.Context, w http.ResponseWriter, params url.Values) {
+	switch params.Get("artifact") {
+	case "buildlog":
+		iid := params.Get("invocation_id")
+		if iid == "" {
+			log.Warningf("Build log requested with empty invocation id.")
+			http.Error(w, "File not found", http.StatusNotFound)
+		}
+		if _, err := s.env.GetInvocationDB().LookupInvocation(ctx, iid); err != nil {
```

Comment:
Does this work for authenticated invocations? The JWT is being nil'd out on line 1050, so at a glance it seems like the ctx here would effectively be an anonymous user context.

Reply:
Same as above.